python/lsst/pipe/base/task.py

```diff
@@ -122,9 +122,10 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = self.ConfigClass()
             self._taskDict = dict()
-            if log is None:
-                log = pexLog.getDefaultLog()
-            self.log = pexLog.Log(log, self._fullName)
+            if log is not None and isinstance(log, Log):
```

Comment:
`isinstance` is not needed here and we can just remove the check.  What you said makes more sense; even if the check stays it should warn or raise if the wrong type of log is passed in.